File: dsa/BinarySearch.py
arr = [2, 5, 8, 12, 16, 23, 38, 56, 72, 91]


def binary_search(array, low, high, key):
    if high >= 1:
        mid = (low + high) // 2

        # print low, high, mid, array[mid]
        if array[mid] == key:
            print(mid)

        elif array[mid] > key:
            return binary_search(array, 1, mid - 1, key)

        else:
            print(arr[mid+1:high])
            return binary_search(array, mid + 1, high, key)
    else:
        print -1


binary_search(arr, 0, len(arr) - 1, 56)

File: dsa/BinarySearch.cpp
#include <iostream>
using namespace std;

int main()
{
    int array[] = {64, 25, 12, 22, 11};
    int size = sizeof(array) / sizeof(array[0]);
    int find = 12;
    int low = 0;
    int high = size - 1;
    while (low <= high)
    {
        int mid = (low + high) / 2;
        if (array[mid] == find)
        {
            cout << mid << " ";
            break;
        }
        else if (array[mid] < find)
            low = mid + 1;
        else
            high = mid - 1;
    }
}